Collections/script.js

```diff
@@ -1,33 +1,206 @@
+      }
+    }
+    let type = oneWordTab[i]; //towrzymy zmienną która będzie właściwością obiektu
+    sentenceObj[type] = licznik; //do obiektu dodajemy właściwość do której przypisujemy wartość wskazaną w liczniku
+  }
+  return sentenceObj; //po wyjściu z pętli zwracamy gotowy obiekt
+}
+let say =
+  "JavaScript byl dla mnie skomplikowany a teraz JavaScript jest dla mnie znacznie prostszy. Viva Javascript";
+
+console.log(countWordOccurrences(say)); //wywołujemy funkcje
+
 // Zadanie 16.
 
+function toCamelCase(word) {
+  let textLength = word.length; //sprawdzamy jaka jest długość słowa, żeby nie pominąc zadnego znaku przy sprawdzaniu
+  let exSign1 = "-"; //wiemy jakich znaków szukamy więc przypisujemy je do zmiannych
+  let exSign2 = "_";
+  let newWord; //do tej zmiannej będziemy przypisywać zmodyfikowane wartości
+
+  for (let i = 0; i < textLength; i++) {
+    //ruszamy z pętą o długości przeszukiwanego stringa
+    if (word.charAt([i]) == exSign1) {
+      //szukamy w pętli czy dany znak występuje na którymś indeksie
+      newWord = word.replace(
+        word.charAt(word.indexOf(exSign1) + 1),
+        word.charAt(word.indexOf(exSign1) + 1).toUpperCase()
+      ); //jeżeli taki znak występuje zamieniamy znak na kolejnym indeksie na wielką literę i przypisujemy do zadeklarowanej wcześniej zmiannej
+
+      word = newWord.replace(exSign1, ""); //do początkowej zmiannej przypisujemy nową zmienną usuwając równocześnie niepotrzebny znak
+    }
+    if (word.charAt([i]) == exSign2) {
+      //dalej robimy dokładnie to samo z kolejnym znakiem
+      newWord = word.replace(
+        word.charAt(word.indexOf(exSign2) + 1),
+        word.charAt(word.indexOf(exSign2) + 1).toUpperCase()
+      );
+
+      word = newWord.replace(exSign2, "");
+    }
+  }
+  console.log(word); //po wyjściu z pętli wyświetlamy w konsoli zmodyfikowanego stringa
+
+  return;
+}
+toCamelCase("java_script"); //wywołujemy finkcje
+toCamelCase("java-script");
+toCamelCase("Java-Script");
+toCamelCase("asp_Net_Core");
+
 // Zadanie 17.
+
+let tablica = [
+  {
+    id: 1,
+    name: "Jacek",
+    grades: [5, 3, 4, 2, 5, 5],
+  },
+  {
+    id: 2,
+    name: "Ewa",
+    grades: [2, 3, 3, 3, 2, 5],
+  },
+  {
+    id: 3,
+    name: "Zygmunt",
+    grades: [2, 2, 4, 4, 3, 3],
+  },
+];
+
+function getStudentTopGrades(tab) {
+  let maxGrades = []; //nowa tablica do której przekażemy wyszukane wartości
+
+  for (let i = 0; i < tab.length; i++) {
+    //uruchamiamy pętle o długości przekazanej tablicy
+    let maxGradesTemp = Math.max.apply(Math, tab[i].grades); //tworzymy zmienną do której przypisujemy maksymalną wartość właściwości obiektu. Używamy funkci applay dedykowanej do tablic
+    maxGrades.push(maxGradesTemp); //wyszukaną wartośc dodajemy do wcześniej utworzonej tablicy
+  }
+  return maxGrades; //funkcja zwraca nową tablicę
+}
+
+console.log(getStudentTopGrades(tablica)); //uruchamiamy finkcję wyświrtlamy w konsoli
```

Comment:
Powyższa funkcja nie radzi sobie z tą częścią zadania:
`Jeśli uczeń nie ma ocen, to przyjmijmy, że jego najwyższa ocena jest równa 0.`